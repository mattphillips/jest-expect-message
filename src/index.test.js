import './';

describe('jest-expect-message', () => {
  test('should fail with custom message', () => {
    expect(() => expect(false, 'Woah this should be false!').toBeTruthy()).toThrowErrorMatchingSnapshot();
  });

  test('should fail without custom message', () => {
    expect(() => expect(false).toBeTruthy()).toThrowErrorMatchingSnapshot();
  });

  test('should pass when given custom message', () => {
    expect(1, 'should have been 1').toBe(1);
  });

  test('should pass when not given custom message', () => {
    expect(1).toBe(1);
  });

  describe('supports custom matchers registered after jest-expect-message', () => {
    expect.extend({
      toBeDivisibleBy(received, argument) {
        const pass = received % argument == 0;
        const message = pass
          ? () => `expected ${received} not to be divisible by ${argument}`
          : () => `expected ${received} to be divisible by ${argument}`;
        return { message, pass };
      }
    });

    it('allows new custom matchers to use custom messages', () => {
      expect(() => expect(100, '100 % 3 === 1').toBeDivisibleBy(3)).toThrowErrorMatchingSnapshot();
      expect(() => expect(101, '101 % 1 === 0').not.toBeDivisibleBy(1)).toThrowErrorMatchingSnapshot();
    });

    it('supports custom asymmetric matchers', () => {
      expect({ apples: 6, bananas: 3 }).toEqual({
        apples: expect.toBeDivisibleBy(1),
        bananas: expect.not.toBeDivisibleBy(2)
      });
    });
  });
});
