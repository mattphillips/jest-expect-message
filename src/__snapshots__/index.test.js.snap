// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-expect-message should fail with custom message 1`] = `
"Custom message:
  Woah this should be false!

[2mexpect([22m[31mreceived[39m[2m).toBeTruthy([22m[2m)[22m

Received: [31mfalse[39m"
`;

exports[`jest-expect-message should fail without custom message 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeTruthy([22m[2m)[22m

Received: [31mfalse[39m"
`;

exports[`jest-expect-message supports custom matchers registered after jest-expect-message allows new custom matchers to use custom messages 1`] = `
"Custom message:
  100 % 3 === 1

expected 100 to be divisible by 3"
`;

exports[`jest-expect-message supports custom matchers registered after jest-expect-message allows new custom matchers to use custom messages 2`] = `
"Custom message:
  101 % 1 === 0

expected 101 not to be divisible by 1"
`;
